spec/unit/provider/package/pip3_spec.rb

```diff
@@ -0,0 +1,257 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+
+provider_class = Puppet::Type.type(:package).provider(:pip3)
+osfamilies = { ['All', nil] => 'pip3' }
+
+describe provider_class do
+
+  before do
+    @resource = Puppet::Resource.new(:package, "fake_package")
+    @provider = provider_class.new(@resource)
+    @client = stub_everything('client')
+    @client.stubs(:call).with('package_releases', 'real_package').returns(["1.3", "1.2.5", "1.2.4"])
+    @client.stubs(:call).with('package_releases', 'fake_package').returns([])
+    XMLRPC::Client.stubs(:new2).returns(@client)
```

Comment:
My bad, I didn't realize the pip(3) providers used XMLRPC. I thought that was left over from when puppet did.